option/option.go

```diff
@@ -0,0 +1,242 @@
+		if !ok {
+			panic("example value must be a boolean")
+		}
+	}
+	if openapiParam.Schema.Value.Type.Is("string") {
+		_, ok := exampleValue.(string)
+		if !ok {
+			panic("example value must be a string")
+		}
+	}
+	return exampleValue
+}
+
+// Registers a parameter for the route. Prefer using the [Query], [QueryInt], [Header], [Cookie] shortcuts.
+func Param(name string, options ...func(*fuego.OpenAPIParam)) func(*fuego.BaseRoute) {
+	param := fuego.OpenAPIParam{
+		Name: name,
+	}
+	// Applies options to fuego.OpenAPIParam
+	for _, option := range options {
+		option(&param)
+	}
+
+	// Applies fuego.OpenAPIParam to openapi3.Parameter
+	// Why not use openapi3.NewHeaderParameter(name) directly?
+	// Because we might change the openapi3 library in the future,
+	// and we want to keep the flexibility to change the implementation without changing the API.
+	openapiParam := &openapi3.Parameter{
+		Name:        name,
+		In:          string(param.Type),
+		Description: param.Description,
+		Schema:      openapi3.NewStringSchema().NewRef(),
+	}
+	if param.GoType != "" {
+		openapiParam.Schema.Value.Type = &openapi3.Types{param.GoType}
+	}
+	openapiParam.Schema.Value.Nullable = param.Nullable
+	openapiParam.Schema.Value.Default = panicsIfNotCorrectType(openapiParam, param.Default)
+
+	if param.Required {
+		openapiParam.Required = param.Required
+	}
+	for name, exampleValue := range param.Examples {
+		if openapiParam.Examples == nil {
+			openapiParam.Examples = make(openapi3.Examples)
+		}
+		exampleOpenAPI := openapi3.NewExample(name)
+		exampleOpenAPI.Value = panicsIfNotCorrectType(openapiParam, exampleValue)
+		openapiParam.Examples[name] = &openapi3.ExampleRef{Value: exampleOpenAPI}
+	}
+
+	return func(r *fuego.BaseRoute) {
+		r.Operation.AddParameter(openapiParam)
+		if r.Params == nil {
+			r.Params = make(map[string]fuego.OpenAPIParam)
+		}
+		r.Params[name] = param
+	}
+}
+
+// Tags adds one or more tags to the route.
+func Tags(tags ...string) func(*fuego.BaseRoute) {
+	return func(r *fuego.BaseRoute) {
+		r.Operation.Tags = append(r.Operation.Tags, tags...)
+	}
+}
+
+// Summary adds a summary to the route.
+func Summary(summary string) func(*fuego.BaseRoute) {
+	return func(r *fuego.BaseRoute) {
+		r.Operation.Summary = summary
+	}
+}
+
+// Description adds a description to the route.
+func Description(description string) func(*fuego.BaseRoute) {
+	return func(r *fuego.BaseRoute) {
+		r.Operation.Description = description
+	}
+}
+
+// OperationID adds an operation ID to the route.
+func OperationID(operationID string) func(*fuego.BaseRoute) {
+	return func(r *fuego.BaseRoute) {
+		r.Operation.OperationID = operationID
+	}
+}
+
+// Deprecated marks the route as deprecated.
+func Deprecated() func(*fuego.BaseRoute) {
+	return func(r *fuego.BaseRoute) {
+		r.Operation.Deprecated = true
+	}
+}
+
+// // AddError adds an error to the route.
+// func AddError(code int, description string, errorType ...any) func(*fuego.BaseRoute) {
```

Comment:
What is the plan here?

Also `RequestContent`

Reply:
Yes i have trouble inferring the returned type and we need it for these 2 options

Reply:
Got it. So we will generally keep this the same moving forward?

Reply:
We don't really _need_ it but it is done that way in the current code.

But I can refactor it, I just need to find a way to do it

Reply:
Mmh I think we'll try to delete the old chaining options, for consistency. I hope it's feasible. The new options pattern do not have access to the ReturnType and BodyType. That's the only difference. It could be both a blessing and a curse, in the future.

Reply:
I guess for right now. Leave it as it is. But let's at least get this comment out of here for cleanliness.